jenkinsapi/jenkins.py

```diff
@@ -346,14 +353,30 @@ def install_plugins(self, plugin_list, restart=False,
         if restart and self.plugins.restart_required:
             self.safe_restart(wait_for_reboot=wait_for_reboot)
 
-    def delete_plugin(self, plugin):
+    def delete_plugin(self, plugin, restart=False, wait_for_reboot=False):
+        """
+        Delete a plugin and optionally restart jenkins. Will not delete
+        dependencies.
+        @param plugin: Plugin to be deleted
+        @param restart: Boolean, restart jenkins after plugin deletion
+        """
         if isinstance(plugin, Plugin):
             plugin = plugin.shortName
         del self.plugins[plugin]
+        if restart and self.plugins.restart_required:
```

Comment:
I would be puzzled why Jenkins haven't restart when I explicitly set parameter to do a restart. It would be impossible to find why w/o looking at the library's source code. Would it be better to restart even if `self.plugins.restart_required` returning False?

And if restart parameter set to `False`, but restart is required, it would be nice to put log warning that restart is required, but continue w/o restarting.

Reply:
Like I said: I reused existing code. Maybe split the `restart` flag to a `restart_when_needed` and `force_restart`? Will add a warning.